extensions/QCOM/SPV_QCOM_tile_shading.asciidoc

```diff
@@ -0,0 +1,225 @@
+QCOM tile shading. _x rate_, _y rate_ and _z rate_ are unsigned 32-bit
+integers. _x rate_ and _y rate_ are a power of 2.  Only valid with the *GLCompute*
+<<Execution_Model,Execution Model>>.
+| <<Literal,Literal>> _x rate_ 
+| <<Literal,Literal>> _y rate_ 
+| <<Literal,Literal>> _z rate_ 
+| *TileShadingQCOM*
+|====
+
+Storage Class
+~~~~~~~~~~~~~
+
+Modify Section 3.7, "Storage Class", adding the following row before CallableDataNV:
+
+[cols="1,15,2*2",options="header",width = "100%"]
+|====
+2+^.^| Storage Class | <<Capability,Enabling Capabilities>> | Enabled by Extension
+| 4491 | *TileAttachmentQCOM* +
+Used to specify tile image attachment variables compatible with the *TileShadingQCOM* capability
+| *TileShadingQCOM* | SPV_QCOM_tile_shading
+|====
+
+Builtin
+~~~~~~~
+
+Modify Section 3.21, "BuiltIn", adding the following rows to the Builtin table before BaryCoordNoPerspAMD:
+
+[cols="1,2*3",options="header",width = "100%"]
+|====
+2+^| BuiltIn ^| Enabling Capabilities
+| 4492 | *TileOffsetQCOM* +
+Framebuffer coordinates (x,y) of the top-left texel of the active tile.
+Available only in fragment and compute shaders.
+| *TileShadingQCOM* +
+Also see extension: SPV_QCOM_tile_shading
+| 4493 | *TileDimensionQCOM* +
+Reports the tile dimension (width, height, z).
+(width, height) represents the tile size in pixels for the active tile
+and z is the number of tile layers.
+Available only in fragment and compute shaders.
+| *TileShadingQCOM* +
+Also see extension: SPV_QCOM_tile_shading
+| 4494 | *TileApronSizeQCOM* +
+Size (horizontal, vertical) of the apron for the active tile.
+Available only in fragment and compute shaders.
+| *TileShadingQCOM* +
+Also see extension: SPV_QCOM_tile_shading
+|====
+
+Capabilities
+~~~~~~~~~~~~
+
+Modify Section 3.31, "Capability", adding the following rows to the Capability table after RayTracingKHR:
+
+[cols="1,5,1",options="header",width = "100%"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 4495 | *TileShadingQCOM* +
+To read from tile image variables, to write to tile image variables and
+to sample tile image variables at the specified coordinates.  
+Makes the *TileAttachmentQCOM* storage
+class, the *TileOffsetQCOM*/*TileDimensionQCOM*/*TileApronSizeQCOM* built-in variables,
+and the *NonCoherentTileAttachmentReadQCOM* and *TileShadingRateQCOM* execution modes
+available to the shader.
+| *Shader*
+|====
+
+Validation Rules
+----------------
+
+An OpExtension must be added to the SPIR-V for validation layers
+to check legal use of this extension:
+
+----
+OpExtension "SPV_QCOM_tile_shading"
+----
+
+ * The _x_ and _y_ rate values of the TileShadingRateQCOM execution mode must be powers of 2.
+
+Issues
+------
+
+Revision History
+----------------
+
+[cols="5,15,15,70"]
+[grid="rows"]
+[options="header"]
+|===========================================================
+|Rev|Date|Author|Changes
+|0|2025-04-15|Wooyoung Kim|Initial version
```

Comment:
This should probably be "1", to be consistent with the document header?

```suggestion
|1|2025-04-15|Wooyoung Kim|Initial version
```

Reply:
done